### ISSUE TITLE
v2 Addon spec as default

### ISSUE THREAD

**Issue by kategengler** [ember-cli/ember-cli#9640]
Documenting some of the things that need to be done before v2 spec is the default for new addons, from a discussion with @ef4:

- [ ] prepublish tool that converts from colocated templates to explicit `setComponentTemplate`
- [ ] Recruit from the learning team to write guides on how to create a v2 addon, and when v2 addon should be used vs v1
- [ ] Update ember-cli-update to also update nested test-apps for v2 addons

**Comment by kellyselden** [comment]
Is there any documentation on nested test-apps for v2 addons? cc @ef4

**Comment by ef4** [comment]
Right now I've been using https://github.com/ember-cli/ember-welcome-page/pull/346 as the working example. I just updated it with a list of open TODOs, some of which are needed docs.

**Comment by NullVoxPopuli** [comment]
can there be two blueprints?
I don't really like the welcome-page approach -- but I'm 100% influenced by monorepo usage and personally liking keeping addon deps strictly to what's needed for the addon, and test stuff separate, such as in:
 - https://github.com/NullVoxPopuli/ember-statechart-component/
 - https://github.com/nullVoxPopuli/ember-resources
 - https://github.com/kaliber5/ember-stargate

**Comment by ef4** [comment]
Yes, and @rwjblue has convinced me that we should push for the monorepo version as the default.

The main benefit of not using a monorepo has historically been the ability for people to point their package.json dependencies directly at your repo, but since most v2 addons will have a build step anyway that already doesn't work. So there's not really a downside to using a monorepo.

**Comment by NullVoxPopuli** [comment]
One thing I like most for addon-monorepos where there is only one addon published from them is having a folder at the top level that _is_ the addon. (not hidden in packages)

**Comment by kellyselden** [comment]
I'm glad we are talking about this. I was planning on writing an RFC for monorepo addon + dummy app (as work time allows).
